docs/tutorial.rst

```diff
@@ -111,6 +111,138 @@ Running this on the example above gives us::
 
    [(7, 0), (6, 1), (4, 2), (3, 2), (5, 1), (2, 2), (1, 2), (0, 2)]
 
+
+++++++++++++++++++++++++
+Traversals with networkx
+++++++++++++++++++++++++
+
+Traversals and other network analysis can also be performed using the sizeable
+`networkx <https://networkx.github.io/documentation/stable/index.html>`_
+library. This can be achieved by calling :meth:`Tree.as_dict_of_dicts` to
+convert a :class:`Tree` instance to a format that can be imported by networkx to
+create a graph::
+
+    import networkx as nx
+
+    g = nx.DiGraph(tree.as_dict_of_dicts())
+    print(sorted(g.edges))
+
+::
+
+    [(5, 0), (5, 1), (5, 2), (6, 3), (6, 4), (7, 5), (7, 6)]
+
+++++++++++++++++++
+Traversing upwards
+++++++++++++++++++
+
+We can revisit the above examples and traverse upwards with
+networkx using a depth-first search algorithm::
+
+    import networkx as nx
+
+    g = nx.DiGraph(tree.as_dict_of_dicts())
+    for u in tree.samples():
+        path = [u] + [parent for parent, child, _ in
+                      nx.edge_dfs(g, source=u, orientation="reverse")]
+        print(u, "->", path)
+
+giving::
+
+   0 -> [0, 5, 7]
+   1 -> [1, 5, 7]
+   2 -> [2, 5, 7]
+   3 -> [3, 6, 7]
+   4 -> [4, 6, 7]
+
++++++++++++++++++++++++++++++++++
+Calculating distances to the root
++++++++++++++++++++++++++++++++++
+
+Similarly, we can yield the nodes of a tree along with their distance to the
+root in pre-order in networkx as well::
+
+    import networkx as nx
+
+    for root in tree.roots:
+        g = nx.DiGraph(tree.as_dict_of_dicts())
+        print(nx.shortest_path_length(g, source=root).items())
+
+Running this on the example above gives us the same result as before::
+
+   [(7, 0), (6, 1), (4, 2), (3, 2), (5, 1), (2, 2), (1, 2), (0, 2)]
+
++++++++++++++++++++++++++++++++++++++++
+Finding nearest neighbors
++++++++++++++++++++++++++++++++++++++++
+
+If some samples in a tree are not at time 0, then finding the nearest neighbor
+of a sample is a bit more involved. Instead of writing our own traversal code
+we can again draw on a networkx algorithm.
+Let us start with an example tree with three samples that were sampled at
+different time points::
+
+.. image:: _static/different_time_samples.svg
+   :width: 200px
+   :alt: An example tree with samples taken at different times
+
+.. code-block:: python
+
+    tables = tskit.TableCollection(sequence_length=1.0)
```

Comment:
Would it be easier to do a quick msprime simulation rather than building a tree up using the tables API? Can always print out the tables via ``print(ts.tables)``

Reply:
Yup. I think standard pop size, followed by an exponentially expanding population with two samples in the standard part and one historical sample in the exponential expansion should probably do? Need to check.